beamer/chain.py

```diff
@@ -197,15 +197,14 @@ def _process_requests(self) -> None:
             elif request.is_filled:
                 claim_request(request, self._context)
 
-            # TODO: This should be triggered some GC event and not be done here directly
             elif request.is_withdrawn:
-                self._log.debug("Removing withdrawn request", request=request)
-                to_remove.append(request.id)
+                active_claims = any([claim.is_withdrawn for claim in self._context.claims])
```

Comment:
- You don't need the list comprehension here I think
- Shouldn't we filter for claims belonging to this request
- Isn't the condition backwards? In my opinion it should be `all(claim.is_withdrawn for claim in claims if ...)`

Reply:
Updated this as it makes sense to me in 9245110e362e2379fbf97fd61a199ac27d2e98ab